### PR TITLE
Adds Current and Previous Period OtherLiabilitiesOrAssets on BalanceSheet

### PR DESCRIPTION
**Pull request by ChrisJBurns**
- adds new `OtherLiabilitiesOrAssetsEntity` DB model and `OtherLiabilitiesOrAssets` `REST` model
- updates `CurrentPeriodTransformer` and `PreviousPeriodTransformer` to include mappings for `OtherLiabilitiesOrAssets`
- adds validation around the current and previous periods when entering any `OtherLiabilitiesOrAssets` and correcting the calculation

This PR contains code already approved from PRs #116 & #119 

Resolves SFA-239, SFA-240, SFA-467,  SFA-468

### CODE REVIEW

**Comment by jsimons1989** [comment]
Can you remove the note in the PR description if/when those changes have been completed please?

File: src/main/java/uk/gov/companieshouse/api/accounts/model/entity/CurrentAssetsEntity.java [hunk 1]
```diff
@@ -0,0 +1,47 @@
+package uk.gov.companieshouse.api.accounts.model.entity;
+
+import org.springframework.data.mongodb.core.mapping.Field;
+
+public class CurrentAssetsEntity {
+
+    @Field("total")
+    private Long total;
```
Comment (BSaunders95) [inline src/main/java/uk/gov/companieshouse/api/accounts/model/entity/CurrentAssetsEntity.java]: Can we space these fields please?

File: src/main/java/uk/gov/companieshouse/api/accounts/validation/CurrentPeriodValidator.java [hunk 2]
```diff
@@ -1,48 +1,132 @@
...
+import uk.gov.companieshouse.api.accounts.model.rest.FixedAssets;
+import uk.gov.companieshouse.api.accounts.model.rest.OtherLiabilitiesOrAssets;
+import uk.gov.companieshouse.api.accounts.model.validation.Errors;
+import uk.gov.companieshouse.api.accounts.model.rest.CurrentAssets;
 import uk.gov.companieshouse.api.accounts.model.validation.Errors;
 import uk.gov.companieshouse.api.accounts.model.rest.CurrentPeriod;
 import uk.gov.companieshouse.api.accounts.model.rest.FixedAssets;
 
 import javax.validation.Valid;
+import java.util.Optional;
```
Comment (BSaunders95) [inline src/main/java/uk/gov/companieshouse/api/accounts/validation/CurrentPeriodValidator.java]: This import is included twice

File: src/main/java/uk/gov/companieshouse/api/accounts/validation/CurrentPeriodValidator.java [hunk 3]
```diff
@@ -1,48 +1,132 @@
...
+    private void calculateOtherLiabilitiesOrAssetsNetCurrentAssets(CurrentPeriod currentPeriod, Errors errors) {
+        OtherLiabilitiesOrAssets otherLiabilitiesOrAssets = currentPeriod.getBalanceSheet().getOtherLiabilitiesOrAssets();
+        Long prepaymentsAndAccruedIncome = Optional.ofNullable(otherLiabilitiesOrAssets.getPrepaymentsAndAccruedIncome()).orElse(0L);
+        Long creditorsDueWithinOneYear = Optional.ofNullable(otherLiabilitiesOrAssets.getCreditorsDueWithinOneYear()).orElse(0L);
+
+        Long totalCurrentAssets;
+        if (currentPeriod.getBalanceSheet().getCurrentAssets() != null) {
+            totalCurrentAssets = currentPeriod.getBalanceSheet().getCurrentAssets().getTotal();
+        }
+        Long calculatedTotal = /* totalCurrentAssets + */ prepaymentsAndAccruedIncome - creditorsDueWithinOneYear;
```
Comment (BSaunders95) [inline src/main/java/uk/gov/companieshouse/api/accounts/validation/CurrentPeriodValidator.java]: Should this commented code be here?

**Comment by BSaunders95** [comment]
And the PR description can be updated to remove the Note now 👍

**Pull request merged**